src/sqlite_db.cpp

```diff
@@ -56,7 +56,8 @@ class SQLite::impl
     {
         void operator()(sqlite3* ptr)
         {
-            std::string filename_(sqlite3_db_filename(ptr, "main"));
+            const auto c_filename = sqlite3_db_filename(ptr, "main");
+            std::string filename_((c_filename == nullptr) ? "" : c_filename);
```

Comment:
If it is a nullptr and it you handle it by passing an empty string why do you then try to close the file? Why not just skip that step?

Reply:
It might be an in memory database, if the pointer is already closed, `sqlite3_close` becomes a NOOP